packages/wdio-browserstack-service/src/util.ts

```diff
@@ -1181,6 +1186,50 @@ export function getFailureObject(error: string|Error) {
     }
 }
 
+export function truncateString(field: string, truncateSizeInBytes: number) {
+    try {
+        const bufferSizeInBytes = Buffer.from(GIT_META_DATA_TRUNCATED).length
+
+        const fieldBufferObj = Buffer.from(field)
+        const lenOfFieldBufferObj = fieldBufferObj.length
+        const finalLen = Math.ceil(lenOfFieldBufferObj - truncateSizeInBytes - bufferSizeInBytes)
+        if (finalLen > 0) {
+            const truncatedString = fieldBufferObj.subarray(0, finalLen).toString() + GIT_META_DATA_TRUNCATED
+            return truncatedString
+        }
+    } catch (error) {
+        BStackLogger.debug(`Error while truncating field, nothing was truncated here: ${error}`)
+    }
+    return field
+}
+
+export function getSizeOfJsonObjectInBytes(jsonData: Object) {
+    try {
+        if (jsonData) {
+            const buffer = Buffer.from(JSON.stringify(jsonData))
+
+            return buffer.length
+        }
+    } catch (error) {
+        BStackLogger.debug(`Something went wrong while calculating size of JSON object: ${error}`)
+    }
+
+    return -1
+}
+
+export function checkAndTruncateVCSInfo(gitMetaData: any) {
```

Comment:
return tyoe

Reply:
lets not use `any`

Reply:
I had to define a `type` in this case for git meta data.